payu/telemetry.py

```diff
@@ -0,0 +1,276 @@
+    manifest data"""
+    manifests = {}
+    for mf in experiment.manifest.manifests:
+        manifests[mf] = experiment.manifest.manifests[mf].data
+
+    return {
+        "manifests": manifests
+    }
+
+
+def get_experiment_run_state(experiment) -> Optional[Dict[str, Any]]:
+    """Returns a dictionary of the experiment run state"""
+    info = {
+        'payu_run_id': experiment.run_id,
+        'payu_current_run': experiment.counter,
+        'payu_n_runs':  experiment.n_runs,
+        'payu_job_status': experiment.run_job_status,
+        'payu_start_time': experiment.start_time.isoformat(),
+        'payu_finish_time': experiment.finish_time.isoformat(),
+        'payu_walltime_seconds':
+            (experiment.finish_time - experiment.start_time).total_seconds(),
+        'payu_version': payu.__version__,
+        'payu_path': os.path.dirname(experiment.payu_path),
+        'payu_config': experiment.config,
+        'user_id':  pwd.getpwuid(os.getuid()).pw_name,
+        'payu_control_path': str(experiment.control_path),
+        'payu_archive_path': str(experiment.archive_path),
+    }
+    return info
+
+
+def get_scheduler_run_info(scheduler: Scheduler) -> Dict[str, Any]:
+    """Returns a dictionary of the scheduler job information"""
+    scheduler_job_id = scheduler.get_job_id(short=False)
+    scheduler_info = scheduler.get_job_info()
+
+    info = {}
+    if scheduler_info is not None:
+        info['scheduler_job_info'] = scheduler_info
+        info['scheduler_type'] = scheduler.name
+    return info
+
+
+def get_external_telemetry_config() -> Optional[Dict[str, Any]]:
+    """Loads the external telemetry configuration file.
+    If a valid file does not exist, return None"""
+    # Check path to telemetry config file exists
+    config_dir = Path(os.environ[TELEMETRY_CONFIG])
+    config_path = config_dir / f"{TELEMETRY_CONFIG_VERSION}.json"
+    if not (config_path.exists() and config_path.is_file()):
+        warnings.warn(
+            f"No config file found at {TELEMETRY_CONFIG}: {config_path}. "
+            "Skipping posting telemetry",
+            UserWarning
+        )
+        return None
+
+    # Attempt to read config file
+    try:
+        with open(config_path, 'r') as f:
+            telemetry_config = json.load(f)
+    except json.JSONDecodeError:
+        warnings.warn(
+            "Error parsing json in configuration file "
+            f"at {TELEMETRY_CONFIG}: {config_path}. "
+            "Skipping posting telemetry"
+        )
+        return None
+
+    # Check for required fields in the telemetry configuration
+    for field in TELEMETRY_CONFIG_FIELDS:
+        if field not in telemetry_config:
+            warnings.warn(
+                f"Required field '{field}' not found in configuration file "
+                f"at {TELEMETRY_CONFIG}: {config_path}. "
+                "Skipping posting telemetry"
+            )
+            return None
+
+    return telemetry_config
```

Comment:
```suggestion
    # Check for required fields in the telemetry configuration
    missing_fields = TELEMETRY_CONFIG_FIELDS.keys() - telemetry_config.keys()
    if missing_fields:
        warnings.warn(
            f"Required field(s) '{missing_fields}' not found in configuration file "
            f"at {TELEMETRY_CONFIG}: {config_path}. "
            "Skipping posting telemetry"
        )
        return None

    return telemetry_config
```
If `TELEMETRY_CONFIG_FIELDS` is converted to a `dict` then I'd suggest something like above.